theano/tensor/nnet/abstract_conv.py

```diff
@@ -1574,8 +1612,15 @@ def make_node(self, img, kern):
 
         if img.type.ndim != 2 + self.convdim:
             raise TypeError('img must be %dD tensor' % (2 + self.convdim))
-        if kern.type.ndim != 2 + self.convdim:
-            raise TypeError('kern must be %dD tensor' % (2 + self.convdim))
+
+        if unshared == True: 
+            if self.convdim == 3:
+                raise NotImplementedError("Unshared 3D convolution not implemented")
+            elif kern.type.ndim != 3 + self.convdim:
+                raise TypeError('kern must be %dD tensor' % (3 + self.convdim))   
```

Comment:
I would swap this: explicitly check that `self.convdim == 2` and raise and error if it is not. The AbstractConv class works for any convdim, even though it's only currently used for 2D and 3D.